conn_test.go

```diff
@@ -751,6 +751,39 @@ func TestContext_Timeout(t *testing.T) {
 	}
 }
 
+func TestWriteCoalescing(t *testing.T) {
+	var buf bytes.Buffer
+	w := &writeCoalescer{
+		w: &buf,
+	}
+	w.cond = sync.NewCond(&w.mu)
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		wg.Done()
+		if _, err := w.write([]byte("one")); err != nil {
+			t.Error(err)
+		}
+	}()
+	wg.Wait()
+
+	wg.Add(1)
+	go func() {
+		wg.Done()
+		if _, err := w.write([]byte("two")); err != nil {
+			t.Error(err)
+		}
+	}()
+	wg.Wait()
+
+	w.flush()
```

Comment:
Considering we test that the write() won't finish until we flash, maybe check that pre-flash the `len(buf)` is still 0?